drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,458 @@
+{
+	if (mux < MAX9611_CONF_IN_COM_MODE)
+		dev->capture_type = MAX9611_DATA_CSA;
+
+	else if (mux == MAX9611_CONF_IN_COM_MODE)
+		dev->capture_type = MAX9611_DATA_RSP;
+
+	else if (mux == MAX9611_CONF_OUT_ADC)
+		dev->capture_type = MAX9611_DATA_OUT;
+
+	else if (mux == MAX9611_CONF_SET_ADC)
+		dev->capture_type = MAX9611_DATA_SET;
+
+	else if (mux == MAX9611_CONF_TEMP)
+		dev->capture_type = MAX9611_DATA_TMP;
+
+	return max9611_reg_update(dev, MAX9611_CTR1, mux, MAX9611_MUX_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get the internal MUX settings
+ * @param dev 		  - The device structure.
+ * @param mux 		  - Pointer to store the obtained mux config
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_mux(struct max9611_dev *dev, enum max9611_mux_conf *mux)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR1, 1, &temp);
+	if (ret)
+		return ret;
+
+	*mux = temp;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Device shutdown or power up
+ * @param dev 		  - The device structure.
+ * @param is_shdn	  - Set to true to shutdown device, false otherwise
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_shutdown(struct max9611_dev *dev, bool is_shdn)
+{
+	if (is_shdn)
+		return max9611_reg_update(dev, MAX9611_CTR1, MAX9611_SHDN_MASK,
+					  MAX9611_SHDN_MASK);
+	else
+		return max9611_reg_update(dev, MAX9611_CTR1, 0, MAX9611_SHDN_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Configure LR bit operation
+ * @param dev 		  - The device structure.
+ * @param is_normal	  - Set true for normal operation. False otherwise (refer to LR bit in datasheet)
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_lr(struct max9611_dev *dev, bool is_normal)
+{
+	if (is_normal)
+		return max9611_reg_update(dev, MAX9611_CTR1, 0, MAX9611_LR_MASK);
+	else
+		return max9611_reg_update(dev, MAX9611_CTR1, MAX9611_LR_MASK,
+					  MAX9611_LR_MASK);
+
+}
+
+/***************************************************************************//**
+ * @brief Get LR bit value
+ * @param dev 		  - The device structure.
+ * @param lr	      - Pointer to store LR bit value
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_lr(struct max9611_dev *dev, bool *lr)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR1, 1, &temp);
+	if (ret)
+		return ret;
+
+	*lr = (bool)((temp & MAX9611_LR_MASK) >> no_os_find_first_set_bit(
```

Comment:
ditto:  `no_os_field_prep` / `no_os_field_get`

Reply:
Resolved in latest commit.